aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotSslUtility.java

```diff
@@ -52,11 +56,23 @@ private AWSIotSslUtility() {
      * @throws KeyManagementException when SSL context cannot be created by key
      *             manager.
      */
+    protected static final String[] ALPN_EXTENSION = {"x-amzn-mqtt-ca"};
+    private static final String TLS_V_1_2 = "TLSv1.2";
+
     public static SSLSocketFactory getSocketFactoryWithKeyStore(KeyStore keyStore)
             throws NoSuchAlgorithmException, UnrecoverableKeyException, KeyStoreException,
             KeyManagementException {
 
-        SSLContext context = SSLContext.getInstance("TLSv1.2");
+        Security.addProvider(new OpenSSLProvider());
```

Comment:
Shouldn't this be done only the ALPN case?

Reply:
Discussed offline. We want to single Provider for all the cases and IoT supports ALPN for both 443 and 8883